src/main/java/io/odpf/firehose/sink/common/AbstractSinkWithDlqProcessor.java

```diff
@@ -0,0 +1,30 @@
+package io.odpf.firehose.sink.common;
+
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.consumer.MessageWithError;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.AbstractSink;
+import io.odpf.firehose.sinkdecorator.dlq.DlqWriter;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public abstract class AbstractSinkWithDlqProcessor extends AbstractSink {
```

Comment:
Maybe we don't need this class. This class doesn't have any abstract methods. We can move this code to AbstractSink

Reply:
this abstract class made the subclasses need to supply and utilise the supplied `DLQWriter` to process DLQ`ed messages rather than relying on `SinkWithDlq` class 
```
    public List<MessageWithError> processDlq(List<MessageWithError> messages) throws IOException {
        return dlqWriter.writeWithError(messages);
    }
```
merging this to `AbstractSink` will enforce other `Sink` to use the same approach, is it okay ?

Reply:
if there are no abstract methods, the class should not be abstract in the first place..
you need to find a better solution.

Reply:
removed this